src/plugins/intel_cpu/src/node.cpp

```diff
@@ -316,6 +315,118 @@ void Node::selectPreferPrimitiveDescriptor(const std::vector<impl_desc_type>& pr
     selectPrimitiveDescriptorByIndex(0);
 }
 
+bool Node::checkScalarShape(const ov::PartialShape& pshape) {
```

Comment:
can we use an existing api instead?
Something like:
``` cpp
  return one_of(ov::shape_size(pshape), 0, 1);
```
There is chance we need to get Shape from a PartialShape first.

Reply:
For example: {1,1,10} also is Scalar shape, `one_of(ov::shape_size(pshape), 0, 1)` seems can't meet our requirement.

Reply:
`For example: {1,1,10} also is Scalar shape`
I believe it is not.
Currently the following shapes are threated as scalar ones:
- {}
- {1}
- {1, 1, 1, 1, ...}

Considering the fact, that you are looking for '{1,1,10}' shapes as well, now I am a bit confused what we are trying to achieve by checking for this type of shapes and estimating them as `1`. Could you please clarify?

Reply:
So confuse you.
Scalar shape self means only 1 element, but actually I want to find out 1 dim shape. I think the reorder of 1 dim shape also have less computation, because It will copy elements sequentially. What do you think? @EgorDuplensky 

1: Keep current logic, just rename scalar shape to 1 dim shape.
2: Exclude 1 dim shape, just check Scalar shape.(of course, it also can fix Zoom's regression issue)
![image](https://github.com/user-attachments/assets/d25eb164-61fb-48cf-b96a-eeecc387871e)

2*(1*256*1*1) < (1*256*64*64)

Reply:
Let's not use a term 'Scalar shape' for the tensor such as {1, 1, 1, 256}
This is misleading and confusing.

I just don't understand why we can't simply remove the special handling for such cases then?
I mean, in the example you have provided above, the generic logic which just computes `ov::shape_size()` would have also worked, isn't it?
Are there any other scenarios where the genetic logic actually would not work?

I can assume that such type of tensors ({1,1,X,1} or {1,1,1,X}) actually never require reorders, if the only difference in the descriptor is layout, so that precision is the same. That is why we might want to treat them differently. But if it is the case, then we better to implement some kind of check isReorderRequired(desc1, desc2), where we return false for such cases and an estimate for them would be 0 (not even 1)

Anyway, let's stick with the current implementation.
We can improve it later.

Reply:
You are right, 
1: ov::shape_size() is good and generic logic, it also works. 
2: Add isReorderRequired
Updated.